backend/drm/drm.c

```diff
@@ -228,30 +228,54 @@ static bool drm_connector_swap_buffers(struct wlr_output *output,
 	return true;
 }
 
-static void drm_connector_set_gamma(struct wlr_output *output,
-		uint32_t size, uint16_t *r, uint16_t *g, uint16_t *b) {
+static void fill_empty_gamma_table(uint32_t size,
+		uint16_t *r, uint16_t *g, uint16_t *b) {
+	for (uint32_t i = 0; i < size; ++i) {
+		uint16_t val = (uint32_t)0xffff * (uint32_t)i / (uint32_t)(size - 1);
```

Comment:
Why does this have so much casting? `i` and `size` are `uint32_t`s anyway.

Reply:
Yeah, true. Fixed.

I kept `uint32_t)0xffff` though.